src/webapi/rendering_context.rs

```diff
@@ -58,4 +1023,319 @@ impl CanvasRenderingContext2d {
+    /// See the CanvasRenderingContext2D.fillText() method to draw the text with the characters filled with color rather than having just their outlines drawn.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/strokeText)
+    // https://html.spec.whatwg.org/#2dcontext:dom-context-2d-stroketext
+    pub fn stroke_text(&self, text: &str, x: f64, y: f64, max_width: Option<f64>) {
+        if let Some(max_width) = max_width {
+            js! { @(no_return)
+                @{&self.0}.strokeText(@{text}, @{x}, @{y}, @{max_width});
+            }
+        }
+        else {
+            js! { @(no_return)
+                @{&self.0}.strokeText(@{text}, @{x}, @{y}, @{Undefined});
+            }
+        }
+    }
+
+    /// Multiplies the current transformation with the matrix described by the arguments of this method. 
+    /// You are able to scale, rotate, move and skew the context.
+    /// See also the setTransform() method which resets the current transform to the identity matrix and then invokes transform().
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/transform)
+    // https://html.spec.whatwg.org/#2dcontext:dom-context-2d-transform
+    pub fn transform(&self, a: f64, b: f64, c: f64, d: f64, e: f64, f: f64) {
+        js! { @(no_return)
+            @{&self.0}.transform(@{a}, @{b}, @{c}, @{d}, @{e}, @{f});
+        }
+    }
+
+    /// Adds a translation transformation by moving the canvas and its origin x horizontally and y vertically on the grid.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/translate)
+    // https://html.spec.whatwg.org/#2dcontext:dom-context-2d-translate
+    pub fn translate(&self, x: f64, y: f64) {
+        js! { @(no_return)
+            @{&self.0}.translate(@{x}, @{y});
+        }
+    }
+
+    fn fill_rule_to_str(&self, fill_rule: FillRule) -> &'static str {
+        match fill_rule {
+            FillRule::NonZero => {
+                "nonzero"
+            }
+
+            FillRule::EvenOdd => {
+                "evenodd"
+            }
+        }
+    }
+}
+
+impl TextMetrics {
+
+    /// Contains the text's advance width (the width of that inline box) in CSS pixels.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics/width)
+    // https://html.spec.whatwg.org/#2dcontext:dom-textmetrics-width
+    pub fn get_width(&self) -> f64 {
+        js! (
+            return @{&self.0}.width;
+        ).try_into().unwrap()
+    }
+}
+
+#[cfg(all(test, feature = "web_test"))]
+mod test {
+    use super::*;
+    use webcore::try_from::TryInto;
+
+    fn new_canvas() -> CanvasRenderingContext2d {
+        js!(
+            var canvas = document.createElement("canvas");
+            return canvas.getContext("2d");
+        ).try_into().unwrap()
```

Comment:
This should use `document().create_element` and `CanvasRenderingContext2d::from_canvas`. (No need to use `js!` macro here.)